### PR TITLE
fix(cli): fix allow_admin allows non-`127.0.0.0/24` to access admin api with empty admin_key

### LINKED ISSUE #9055: bug: If `allow_admin` contains 127.0.0.0/24, requests from other ips without adminKey are allowed by default
### Current Behavior

If `allow_admin` contains 127.0.0.0/24, requests from other ips without the admin key are allowed by default.

https://github.com/apache/apisix/blob/27535e9831b303ef721046bd51e471abb0be9c13/apisix/cli/ops.lua#L190

### Expected Behavior

Only requests from 127.0.0.0/24 without adminKey are allowed. Otherwise, they should be banned if the admin key isn't configured in `config.yaml`.

### Error Logs

_No response_

### Steps to Reproduce

1. use the config below:
```
deployment:
  admin:
    allow_admin:
      - 0.0.0.0/0
      - 127.0.0.0/24
    admin_key:
      # empty
```
2. Requests from other ips without adminKey are allowed.

### Environment

- APISIX version (run `apisix version`): 3.2.0
- Operating system (run `uname -a`): Ubuntu 18.04
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

### PR DESCRIPTION
**Pull request by An-DJ**
### Description

Fixes #9055 

### Checklist

- [X] I have explained the need for this PR and the problem it solves
- [X] I have explained the changes or the new features added to this PR
- [X] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [X] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by moonming — APPROVED** [review]

File: apisix/cli/ops.lua [hunk 1]
```diff
@@ -185,12 +186,9 @@ local function init(env)
...
     local allow_admin = yaml_conf.deployment.admin and
         yaml_conf.deployment.admin.allow_admin
-    if yaml_conf.apisix.enable_admin and allow_admin then
-        for _, allow_ip in ipairs(allow_admin) do
-            if allow_ip == "127.0.0.0/24" then
-                checked_admin_key = true
-            end
-        end
+    if yaml_conf.apisix.enable_admin and allow_admin
+       and table.getn(allow_admin) == 1 and allow_admin[1] == "127.0.0.0/24" then
```
Comment (spacewander) [inline apisix/cli/ops.lua:190]: table.getn is deprecated, use `#` instead

File: t/cli/test_admin.sh [hunk 2]
```diff
@@ -154,6 +154,45 @@ fi
 
 echo "pass: missing admin key and show ERROR message"
 
+# missing admin key, only allow 127.0.0.0/24 to access admin api
+
+git checkout conf/config.yaml
```
Comment (spacewander) [inline t/cli/test_admin.sh:159]: We don't need to checkout the config and then override it

File: t/cli/test_admin.sh [hunk 3]
```diff
@@ -154,6 +154,45 @@ fi
...
+  admin:
+    allow_admin:
+      - 127.0.0.0/24
+    allow_admin: ~
+' > conf/config.yaml
+
+make init > output.log 2>&1 | true
+
+grep -E "ERROR: missing valid Admin API token." output.log > /dev/null
+if [ ! $? -ne 0 ]; then
```
Comment (spacewander) [inline t/cli/test_admin.sh:172]: The condition here is incorrect?

File: t/cli/test_admin.sh [hunk 4]
```diff
@@ -154,6 +154,43 @@ fi
...
+  admin:
+    allow_admin:
+      - 127.0.0.0/24
+    allow_admin: ~
+' > conf/config.yaml
+
+make init > output.log 2>&1 | true
+
+grep -E "ERROR: missing valid Admin API token." output.log > /dev/null
+if [ $? -eq 0 ]; then
```
Comment (spacewander) [inline t/cli/test_admin.sh:170]: The condition here is still incorrect, please think again what is going to check in this test.

**Review by monkeyDluffy6017 — APPROVED** [review]

File: t/cli/test_admin.sh (hunk 4 above)
Comment (monkeyDluffy6017) [reply t/cli/test_admin.sh]: I have checked with @spacewander, the condition is correct

Comment (spacewander) [reply t/cli/test_admin.sh]: @monkeyDluffy6017 
Yes, you are right. Sorry for the wrong suggestion.

**Review by spacewander — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

**Review by moonming — APPROVED** [review]

**Review by leslie-tsang — APPROVED** [review]

**Pull request merged**